python/its-interqueuemanager/its_iqm/iqm.py

```diff
@@ -33,28 +52,19 @@ def __init__(
             password=cfg["local"]["password"],
             client_id=cfg["local"]["client_id"],
             local_qm=None,
-            prefix=cfg["general"]["prefix"],
-            suffix=cfg["general"]["suffix"],
-            copy_from="inQueue",
-            copy_to=[
-                "outQueue",
-                cfg["local"]["interqueue"],
-            ],
+            copy_from=inqueue,
+            copy_to=[outqueue, interqueue],
         )
 
         # The central authority will call our update_cb(), for which we
         # will need to have a valid local_qm to pass to the neighbours
         # queue managers, so we need to handle the central authority
         # after we create the local QM.
-        authority_type = self.cfg["authority"]["type"]
-        if authority_type == "file":
-            self.authority = its_iqm.authority.file.Authority(self.cfg, self.update_cb)
-        elif authority_type == "http":
-            self.authority = its_iqm.authority.http.Authority(self.cfg, self.update_cb)
-        elif authority_type == "mqtt":
-            self.authority = its_iqm.authority.mqtt.Authority(self.cfg, self.update_cb)
-        else:
-            raise ValueError(f"unknown central authority type {central_type}")
+        self.authority = its_iqm.authority.Authority(
+            self.instance_id,
+            self.cfg["authority"],
+            self.update_cb,
+        )
 
     def run_forever(self):
         self.neighbours = dict()
```

Comment:
do it into __init__ if it has got a class scope (bad lazy loading in my opinion).

Reply:
It does not have _class_ scope, but _instance_ scope. Also, it is not needed unless
the instance is actualy "running", so it should only be initialised when starting
the instance. Finally, if we restart the instance (not possible with the `main()`
we currently have, but there is otherwise nothing that would prevent it), we want
to start afresh with an empty set of neighbours, so we still do need to initialise
it at the beginning of `run_forever()`.